learn/hoon/hoon-tutorial/egg-timer.md

```diff
@@ -0,0 +1,120 @@
+
+The sample of the `door` is:
+
+```
+[bowl:gall ~]
+```
+
+You can find the full definition of `bowl` in `sys/zuse.hoon`, but for now it's enough to know that this is the default app state and includes various faces for information. Below are some important such faces:
+
+`our`  The current ship
+`src`  The ship the request originated with
+`ost`  A reference to a chain of events
+`eny`  Entropy
+`now`  The current time of the event
+
+The `door` we've made has two arms `poke-noun` and `wake`. Gall is capable of dispatching `pokes`, or requests, to an app based on the mark of the data given along with that poke. These are sent to the arm with the name that matches the mark of the data. Here we use the generic `noun` mark:
+
+```
+++  poke-noun
+  |=  t=@dr
+  ^+  [*(list effect) +>.$]
+  :_  +>.$  :_  ~
+  [ost %wait /(scot %da now) (add now t)]
+```
+
+In the above code, we create a gate that takes a single `@dr` argument. `@dr` is an aura for a 128-bit relative date. Here are a few examples.
+
+`~s17`  17 seconds
+`~m20`  20 minutes
+`~d42`  42 days
+
+As a matter of good type hygiene, we explicitly cast the output of this gate with `^+` to ensure we are producing the correct thing for Gall to handle. `^+` is the rune for casting by example. Our example is a cell: `list` of `effect`, which we `bunt` with `*`, is the head; `+>.$`, the enclosing core which is our `door`, is the tail.
+
+Next we're going to use the `:_` rune which is just the inverted form a `:-` the cell construction rune. We use it twice so the actual data will end up looking something like:
+
+```
+[[[ost %wait /(scot %da now) (add now t)] ~] +>.$]
+```
+
+`ost` is the `bone`, the opaque reference to a chain of events, that comes from `bowl`, so we're going to continue to use it here. `%wait` is the name of the `syscall`, in this case a request to the Behn vain to start a timer for us.
+
+After `%wait`, we have the `path`, a `list` of `cords` that serves as the unique identifier for this `syscall`:
+
+```
+/(scot %da now)
+```
+
+The `/` here is an irregular syntax to make a `path`, and `scot` is going to transform `now`, the current time as given to us by `bowl`, from a `@da` to a `cord`.
+
+The final part of this `syscall` is:
+
+```
+(add now t)
+```
+
+`now` is the current time of type `@da`, and `t` was declared as `@dr`. Because they are both atoms, we can add `now` and `t` these two to get an atom that is `t` units of time into the future from `now`. That produced atom can be interpreted as a `@da`.
+
+That's all for our `poke-noun` arm. But what about when the timer goes off? Behn will create a `move` similar to how we created one with `effect`, only this time it will end up being dispatched back to us in the `++wake` arm. Behn always looks for an arm named `++wake` when a timer goes off; it is defined this way in the vane's code. So any So any app that wants to use a timer trigger needs to have an arm called `++wake`.
```

Comment:
Technically incorrect in a couple ways, but I don't think we care much here?

Reply:
We absolutely do care. Please elaborate.

Reply:
> Behn will create a `move`
> Behn always looks for an arm named `++wake`
> it is defined this way in the vane's code.

The moves our application sends out don't go directly to the vanes. They first go through Gall, which then sends them off to the vanes (after stripping off the wire, which the vanes don't care about) in the form of vane `task`s.

If/when a vane wants to respond or otherwise talk to our application, it sends a `gift` back up the `duct` (aka call stack), so that ends up back in Gall. Gall then (after tacking the wire back on, because apps do care about it), figures out the appropriate app arm to call based on the `gift` contents, and attempts calling it.

I imagine that goes deeper into Gall/vanes than this particular doc cares about, so being a bit hand-wave-y here is fine. But, as it stands, this may perhaps be a bit misleading.

Reply:
No, I think this does need some modification. I just misunderstood what vanes do in response to requests. Thanks. I'll make a pr.